client/shipping-label/state/reducer.js

```diff
@@ -328,25 +329,41 @@ reducers[ OPEN_ADD_ITEM ] = ( state ) => {
 	return {
 		...state,
 		showAddItemDialog: true,
+		addedItems: {},
 	};
 };
 
 reducers[ CLOSE_ADD_ITEM ] = ( state ) => {
 	return {
 		...state,
-		movedItemIndex: -1,
 		showAddItemDialog: false,
 	};
 };
 
-reducers[ SET_ADDED_ITEM ] = ( state, { sourcePackageId, movedItemIndex } ) => {
+reducers[ SET_ADDED_ITEM ] = ( state, { sourcePackageId, movedItemIndex, added } ) => {
 	return {
 		...state,
-		sourcePackageId,
-		movedItemIndex,
+		addedItems: { ...state.addedItems,
+			[ sourcePackageId ]: { ...state.addedItems[ sourcePackageId ],
+				[ movedItemIndex ]: added,
+			}
+		},
 	};
 };
 
+reducers[ ADD_ITEMS ] = ( state, { targetPackageId } ) => {
+	return _.reduce( state.addedItems, ( prevState, inPackage, originPackageId ) => {
+		const movedItemIndices = _.toPairs( inPackage )
+			.filter( ( [ , added ] ) => added )
+			.map( ( [ movedItemIndex ] ) => +movedItemIndex );
+		return _.sortBy( movedItemIndices ).reverse().reduce( ( prevPrevState, movedItemIndex ) => {
+			return reducers[ MOVE_ITEM ]( prevPrevState, { originPackageId, movedItemIndex, targetPackageId } );
+		}, prevState );
+	}, { ...state,
+		showAddItemDialog: false,
+	} );
+};
```

Comment:
As discussed in Slack, it's not immediately clear what's going on in this reducer.

Adding some comments to explain lodash function usage would help explain things. For instance, what's the purpose of using `_.sortBy()` with no second argument, and why does the result need `reverse()`ing?